lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -229,12 +233,22 @@ private JourneyResponse getJourneyResponse(
             AuditEventUser auditEventUser =
                     new AuditEventUser(userId, ipvSessionId, govukSigninJourneyId, ipAddress);
 
-            var vcs = getVerifiableCredentials(userId, clientOAuthSessionItem.getEvcsAccessToken());
+            var evcsAccessToken = clientOAuthSessionItem.getEvcsAccessToken();
+            var vcs = getVerifiableCredentials(userId, evcsAccessToken);
+            var pendingReturnVcs = vcs.stream().filter(vc -> vc.isPendingReturn()).toList();
```

Comment:
I feel a bit uncomfortable about this - this won't work if the VCs have been read from anywhere except EVCS. In the current state it'll work because we only care about the result if EVCS reads are enabled, but it seems hard to reason about.

I wonder if we actually want EVCS to return some other data structure when you read, e.g. a `Map<EvcsVCState, List<VerifiableCredential>>`, or some other wrapper class? That way it's really clear which set of VCs you're looking at?

Otherwise I think we need to make it clear that the `isPending()` method _only works_ if you're using VCs read from EVCS (e.g. failing hard if you try and use it in any other scenario).

Reply:
Yes ok I like the EVCS Map idea - will look into it :)

Reply:
I've added a combo of he Map function and an inner class to keep track of the properites (hasF2f / hasPending)